### PR TITLE
Proposal for the introduction of extensions

### PR DESCRIPTION
**Pull request by FFroehlich**
This pull request describes a proposal for the extension of PEtab (see #528) that allows changes to the format through extensions. The proposed development and acceptance procedure is described in the changes to development.rst . The other change the documentation to indicate that the PEtab YAML file becomes mandatory.

Required steps for changes to the PEtab specification
-----------------------------------------------------
- [x] Updated format specifications (this PR)
- [x] Updated converter (no changes necessary)
- [x] Updated validator (no changes necessary)
- [x] Updated test suite (no changes necessary)
- [x] Updated changelog (this PR)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/petab-extensions.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+name: PEtab Extension
+about: Suggest a new extension for PEtab core
+title: ''
+labels: file format
+assignees: ''
+
+---
+
+**Name of the Extension**
+Please make sure that the extension is named matches the regular expressin `^[a-zA-Z_]\w*$`.
```
Comment (dweindl) [inline .github/ISSUE_TEMPLATE/petab-extensions.md:11]:
```suggestion
Please make sure that the extension name matches the regular expression `^[a-zA-Z_]\w*$`.
```

File: .github/ISSUE_TEMPLATE/petab-extensions.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+**Describe the solution you would like**
+A clear and concise description of the changes you want to propose. Please describe any additional fields / files you would want to add, including allowed inputs and implications.
+
+**Describe why this should not be implemented by changes to PEtab core**
+A clear and concise description in what way the proposed changes introduce features that are orthogonal to the PEtab core specification.
+
+**List the extension library that implements validation checks**
+A link to the website or github repository that accompanies the proposed extension.
+
+**List two toolboxes that support the proposed standard **
```
Comment (dweindl) [inline .github/ISSUE_TEMPLATE/petab-extensions.md:25]:
```suggestion
**List two toolboxes that support the proposed standard**
```

File: README.md [hunk 3]
```diff
@@ -125,6 +125,8 @@ will have to:
 
 1. Create a parameter table.
 
+1. Create a yaml file that list the model and all of the tables above.
```
Comment (dweindl) [inline README.md:128]:
```suggestion
1. Create a yaml file that lists the model and all of the tables above.
```

File: doc/development.rst [hunk 4]
```diff
@@ -192,6 +192,59 @@ Upon a new release, the PEtab editors ensure that
...
+Requirements for new extensions:
+
+* Specifications in PEtab extensions take precedence over PEtab core, i.e., they
+can ease or refine format restrictions imposed by PEtab core.
+* PEtab extensions should extend PEtab core with new orthogonal features or
+tasks, i.e., they should not make trivial changes to PEtab core.
+* PEtab extensions must be named according to ^[a-zA-Z_]\w*$
+* PEtab extensions must be versioned using semantic versioning.
+* PEtab extensions required for interpretation of a problem specification must
+be specified in the in the PEtab-YAML files
```
Comment (dweindl) [inline doc/development.rst:217]:
```suggestion
be specified in the PEtab-YAML files
```

**Comment by dweindl** [comment]
Here is a draft for the updated yaml schema to account for extensions that I created some time ago: https://github.com/PEtab-dev/libpetab-python/blob/yaml_schema_for_extensions/petab/petab_schema.yaml

File: doc/development.rst [hunk 5]
```diff
@@ -192,6 +192,59 @@ Upon a new release, the PEtab editors ensure that
...
+* The authors provide a library that implements validation checks for the
+proposed format.
+
+Developers are free to develop any PEtab extension. To become an official
+PEtab extension, it needs to go through the following process.
+
+1. The developers write a proposal describing the motivation and specification
+of the extension, following the respective issue template provided in this
+repository.
+1. The proposal submitted as an issue in this repository.
```
Comment (yannikschaelte) [inline doc/development.rst:228]:
```suggestion
1. The proposal is submitted as an issue in this repository.
```

File: doc/development.rst [hunk 6]
```diff
@@ -192,6 +192,59 @@ Upon a new release, the PEtab editors ensure that
...
+1. The technical specification and documentation of the extension is submitted
+as pull request in this repository that references the respective issue.
```
Comment (yannikschaelte) [inline doc/development.rst:230]:
```suggestion
as a pull request in this repository that references the respective issue.
```

File: doc/documentation_data_format.rst [hunk 7]
```diff
@@ -58,6 +58,9 @@ and
 - A parameter file specifying optimization parameters and related information
   [TSV]
 
+- A grouping file that list all of the files and provides additional information
```
Comment (yannikschaelte) [inline doc/documentation_data_format.rst:61]:
```suggestion
- A grouping file that lists all of the files and provides additional information
```

**Review by yannikschaelte — APPROVED** [review]

**Pull request closed without merging**